updater/fetchers/nodejs/nodejs.go

```diff
@@ -0,0 +1,182 @@
+		log.Errorf("could not download Nodejs's update: %s", err)
+		return resp, cerrors.ErrCouldNotDownload
+	}
+	defer r.Body.Close()
+
+	// Get the latest date of the latest update's JSON data
+	latestUpdate, err := datastore.GetKeyValue(updaterFlag)
+	if err != nil {
+		return resp, err
+	}
+
+	// Unmarshal JSON.
+	var advisories nodejsAdvisories
+	if err = json.NewDecoder(r.Body).Decode(&advisories); err != nil {
+		log.Errorf("could not unmarshal Nodejs's JSON: %s", err)
+		return resp, cerrors.ErrCouldNotParse
+	}
+
+	resp.Vulnerabilities, resp.FlagValue = parseNodejsAdvisories(advisories.Results, latestUpdate)
+	resp.FlagName = updaterFlag
+
+	return resp, nil
+}
+
+func parseNodejsAdvisories(advisories []nodejsAdvisory, latestUpdate string) (vulnerabilities []database.Vulnerability, newUpdated string) {
+	mvulnerabilities := make(map[string]*database.Vulnerability)
+
+	for _, advisory := range advisories {
+		if latestUpdate >= advisory.UpdatedAt {
+			break
+		}
+		if advisory.UpdatedAt > newUpdated {
+			newUpdated = advisory.UpdatedAt
+		}
+		for _, vulnName := range advisory.CVES {
+			// Get or create the vulnerability.
+			vulnerability, vulnerabilityAlreadyExists := mvulnerabilities[vulnName]
+			if !vulnerabilityAlreadyExists {
+				vulnerability = &database.Vulnerability{
+					Name:        vulnName,
+					Link:        cveURLPrefix + strings.TrimLeft(vulnName, "CVE-"),
+					Severity:    types.Unknown,
+					Description: advisory.Overview,
+				}
+			}
+
+			// Set the priority of the vulnerability.
+			// A vulnerability has one urgency per advisory it affects.
+			// The highest urgency should be the one set.
+			if urgency := types.ScoreToPriority(advisory.CvssScore); urgency.Compare(vulnerability.Severity) > 0 {
+				vulnerability.Severity = urgency
+			}
+
+			// Create and add the feature version.
+			pkg := database.FeatureVersion{
+				Feature: database.Feature{
+					Name: advisory.ModuleName,
+					Namespace: database.Namespace{
+						Name: "nodejs:" + defaultNodejsVersion,
+					},
+				},
+			}
+			if version, err := getAdvisoryVersion(advisory.PatchedVersions); err == nil {
+				pkg.Version = version
+			}
+			vulnerability.FixedIn = append(vulnerability.FixedIn, pkg)
+
+			// Store the vulnerability.
+			mvulnerabilities[vulnName] = vulnerability
+		}
+	}
+
+	// Convert the vulnerabilities map to a slice
+	for _, v := range mvulnerabilities {
+		vulnerabilities = append(vulnerabilities, *v)
+	}
+
+	return
+}
+
+// getAdvisoryVersion parses a string containing one or multiple version ranges
+// and returns upper-bound. By nature, this simplification may lead to false-positives
+func getAdvisoryVersion(fullVersion string) (types.Version, error) {
+	fixedVersion := types.MinVersion
+
+	for _, version := range strings.Split(fullVersion, "||") {
+		ovs := getOperVersions(version)
+		for _, ov := range ovs {
+			if ov.Oper == ">" {
+				if curVersion, err := types.NewVersion(ov.Version + defaultVersionSuffix); err != nil {
```

Comment:
I think a thorough solution is changing the current 'version' from types.NewVersion to a full  description string like [">=2.5.0 <= 3.0.0 || >=3.1.0"](https://api.nodesecurity.io/advisories/92). It could solve both this issue and [false-positive issue](https://github.com/liangchenye/clair/blob/npm/updater/fetchers/nodejs/nodejs.go#L153).

It may not necessary to [System package managers]( https://github.com/theupdateframework/tuf#what-is-a-software-update-system), but useful for 'Application updaters' and 'Library package managers'.